src/object.d

```diff
@@ -1890,6 +1891,16 @@ void* aaLiteral(Key, Value)(Key[] keys, Value[] values) @trusted pure
 
 alias AssociativeArray(Key, Value) = Value[Key];
 
+void clear(T : Value[Key], Value, Key)(T aa)
+{
+    _aaClear(*cast(void **)&aa);
+}
+
+void clear(T : Value[Key], Value, Key)(T* aa)
```

Comment:
The other methods are imitating a builtin property and UFCS doesn't dereference pointers. I'd say keep this for clear as well.